backend/src/services/secret-v2-bridge/secret-v2-bridge-fns.ts

```diff
@@ -415,14 +422,21 @@ export const expandSecretReferencesFactory = ({
     return secretCache[cacheKey][secretKey] || "";
   };
 
-  const recursivelyExpandSecret = async (dto: { value?: string; secretPath: string; environment: string }) => {
-    if (!dto.value) return "";
+  const recursivelyExpandSecret = async (dto: {
```

Comment:
Not necessarily saying to do so here but don't we need at some level to check permissions for each reference within the secret? Otherwise can't people potentially see a secret value they shouldn't via an expanded value/reference tree?